src/tools/mesh_converter.cpp

```diff
@@ -0,0 +1,684 @@
+
+#include "axom/mint.hpp"
+#include "axom/slam.hpp"
+#include "axom/spin.hpp"
+#include "axom/quest.hpp"
+
+#include "axom/CLI11.hpp"
+#include "axom/fmt.hpp"
+
+#include <string>
+
+namespace slam = axom::slam;
+namespace primal = axom::primal;
+namespace quest = axom::quest;
+namespace spin = axom::spin;
+
+/**
+ * This class represents a simplicial mesh
+ * 
+ * It uses slam to encode the vertices and simplices (zones) of a simplicial mesh 
+ * as well as the connectivity relation from the simplices to vertices,
+ * but does not represent other entities (e.g. edges, faces) 
+ * or relations (like adjacency or vertex-simplex incidences).
+ * 
+ * It Can be used to represent both tetrahedral meshes (TOPO_DIM=3) 
+ * and triangle meshes (TOPO_DIM=2).
+ */
+template <int TOPO_DIM, int SPACE_DIM = 3>
+struct SimplicialMesh
+{
+public:
+  static constexpr int NODES_PER_ZONE = TOPO_DIM + 1;
+
+  using PositionType = slam::DefaultPositionType;
+
+  // types for sets
+  using ElementType = slam::DefaultElementType;
+  using PointType = primal::Point<double, SPACE_DIM>;
+  using NodeSet = slam::PositionSet<PositionType, ElementType>;
+  using ZoneSet = slam::PositionSet<PositionType, ElementType>;
+
+  // types for relations
+  using Indir = slam::policies::ArrayIndirection<PositionType, ElementType>;
+  using ZNStride = slam::policies::CompileTimeStride<PositionType, NODES_PER_ZONE>;
+  using ConstantCardinality = slam::policies::ConstantCardinality<PositionType, ZNStride>;
+  using ZoneToNodeRelation =
+    typename slam::StaticRelation<PositionType, ElementType, ConstantCardinality, Indir, ZoneSet, NodeSet>;
+  using ZNBuilder = typename ZoneToNodeRelation::RelationBuilder;
+
+  /// types for maps
+  using Vertices =
+    slam::Map<PointType, NodeSet, slam::policies::ArrayIndirection<PositionType, PointType>>;
+  using VertexArray =
+    typename slam::policies::ArrayIndirection<PositionType, PointType>::IndirectionBufferType;
+  using VertexMapBuilder = typename Vertices::MapBuilder;
+  // Define the SimplexType based on the topological dimension
+  using SimplexType = std::conditional_t<
+    TOPO_DIM == 3,
+    primal::Tetrahedron<double, SPACE_DIM>,
+    std::conditional_t<TOPO_DIM == 2, primal::Triangle<double, SPACE_DIM>, primal::Segment<double, SPACE_DIM>>>;
+
+  static_assert(SPACE_DIM <= 3, "SPACE_DIM can be at most 3");
+  static_assert(TOPO_DIM <= SPACE_DIM, "TOPO_DIM must be less than or equal to SPACE_DIM");
+  static_assert(1 <= TOPO_DIM, "TOPO_DIM must be greater than 0");
+
+public:
+  SimplicialMesh(int num_verts = 0, int num_zones = 0) { reset(num_verts, num_zones); }
+
+  /// Resets the internal sets and relations and ensures we have the right sizes
+  void reset(int num_verts = 0, int num_zones = 0)
+  {
+    nodeSet = NodeSet(num_verts);
+    zoneSet = ZoneSet(num_zones);
+
+    // set up the zone-node relation
+    m_zn.resize(zoneSet.size() * NODES_PER_ZONE);
+    znRel = ZNBuilder().fromSet(&zoneSet).toSet(&nodeSet).indices(
```

Comment:
What does this say?

Reply:
The slam classes are policy-based and can be constructed differently based on how they're configured.
To make it easier to construct them, we use a "builder" interface w/ chained calls that set the desired class members.
The builder class is a member of the relation class (see the type aliases higher up in this file), and the relation has a constructor that takes a builder instance.

In this case, we're building a connectivity relation from the zones to the nodes (specifically, from the set of zone indices to the set of node indices, and we're using the connectivity array as our storage).

Note:  Based on @Arlie-Capps's suggestion, I renamed the class members and encapsulated the class data after your review.